### ISSUE TITLE
_CoroGuard cannot be awaited

### ISSUE THREAD

**Issue by litwisha** [aio-libs/aiohttp#2026]
## Long story short

1) [_CoroGuard](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/helpers.py#L65) class has no implementation for `__await__` method, only `__iter__`. 
That's why`_CoroGuard` instance cannot be `awaited`, just `yielded from`.

2) Fix warning [message](https://github.com/aio-libs/aiohttp/blob/master/aiohttp/client.py#L538) for `ClientSession.close()` method

## Expected behaviour

- `_CoroGuard` can be awaited

## Actual behaviour

- `_CoroGuard` cannot be awaited

## Steps to reproduce

```
#!/usr/bin/env python
import aiohttp
import asyncio


async def coro_function(loop):
    session = aiohttp.ClientSession(loop=loop)
    await session.close()


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    try:
        loop.run_until_complete(coro_function(loop))
    finally:
        loop.close()
```

## Your environment
```
OS X 10.12.3
python 3.6.0
aiohttp==2.2.0
```

**Comment by hellysmile** [comment]
Why it should produce warning if not awaited/yield from? Seems starting aiohttp 2.0+ close is synchronous method, previously it was returning empty future.

**Comment by asvetlov** [comment]
@hellysmile 

It's still asynchronous but I want to stress people and force them to use `await session.close()` during migration period.

The method still returns empty future but will be replaced with pure coroutine eventually.

**Comment by asvetlov** [comment]
Closed by #2027
